stonesoup/proposal/simple.py

```diff
@@ -0,0 +1,141 @@
+    """This proposal uses the kalman filter prediction and update steps to
+    generate new set of particles and weights
+    """
+    predictor: Predictor = Property(
+        doc="predictor to use the various values")
+    updater: Updater = Property(
+        doc="Updater used for update the values")
+
+    def rvs(self, prior: State, measurement: Detection = None, time_interval=None,
+            **kwargs):
+        """Generate samples from the proposal.
+            Use the kalman filter predictor and updater to create a new distribution
+        Parameters
+        ----------
+        state: :class:`~.State`
+            The state to generate samples from.
+        measurement: :class:`~.Detection`
+            the measurement that is used in the update step of the kalman prediction,
+            (the default is `None`)
+        time_interval: :class:`datetime.time_delta`
+            time interval of the prediction is needed to propagate the states
+
+        Returns
+        -------
+        : :class:`~.ParticlePrediction`
+        """
+
+        if measurement is not None:
+            timestamp = measurement.timestamp
+            time_interval = measurement.timestamp - prior.timestamp
+        else:
+            timestamp = prior.timestamp + time_interval
+
+        if time_interval.total_seconds() == 0:
+            return Prediction.from_state(prior,
+                                         parent=prior,
+                                         state_vector=prior.state_vector,
+                                         timestamp=prior.timestamp,
+                                         transition_model=self.predictor.transition_model,
+                                         prior=prior)
+
+        prior_cls = GaussianState  # Default
+        if isinstance(self.predictor, SqrtKalmanPredictor):
+            prior_cls = SqrtGaussianState
+
+        # Null covariance for the particles
+        null_covar = np.zeros_like(prior.covar)
+
+        predictions = [
+            self.predictor.predict(
+                prior_cls(particle_sv, null_covar, prior.timestamp),
+                timestamp=timestamp)
+            for particle_sv in prior.state_vector]
+
+        if measurement is not None:
+            updates = [self.updater.update(SingleHypothesis(prediction, measurement))
+                       for prediction in predictions]
+        else:
+            updates = predictions  # keep the prediction
+
+        # Draw the samples
+        samples = np.array([state.state_vector.reshape(-1) +
+                            mvn.rvs(cov=state.covar).T
+                            for state in updates])
+
+        # Compute the log of q(x_k|x_{k-1}, y_k)
+        post_log_weights = np.array([mvn.logpdf(sample,
+                                                np.array(update.mean).reshape(-1),
+                                                update.covar)
+                                     for sample, update in zip(samples, updates)])
```

Comment:
This would be better to handle custom types, as mvn converts to float arrays before taking difference.

```suggestion
        post_log_weights = np.array([mvn.logpdf(sample - update.mean.reshape(-1), cov=update.covar)
                                     for sample, update in zip(samples, updates)])
```

Also, here you use `update.mean` but above `update.state_vector`. Should these not be both the same? (In theory they should be anyway I guess? but...)

Reply:
I see your point, `mean` and `state_vector` are equal in this case. I changed it